index.js

```diff
@@ -313,4 +386,95 @@ function reducer(acc, cur) {
   return acc;
 }
 
+/**
+ * Given an error from got, see if Shopify told us how long to wait before
+ * retrying. Return a duration in ms if we can, and otherwise return null.
+ *
+ * @param {Object} error Error object from got call
+ * @return {Boolean | null}
+ * @private
+ **/
+function maybeRetryMS(error) {
+  // for simplicity, retry network connectivity issues after a hardcoded 1s
+  if (
+    typeof error === 'object' &&
+    'code' in error &&
+    RetryableErrorCodes.has(error.code)
+  ) {
+    return 1000;
+  }
+
+  const response = error.response;
+  if (!response || !response.headers) {
+    return null;
+  }
+
+  if (response.headers && response.headers['retry-after']) {
+    const value = parseFloat(response.headers['retry-after']);
+    if (isNaN(value)) {
+      const when = new Date(value).valueOf();
+      return when - Date.now();
+    } else if (isFinite(value)) {
+      return value;
+    } else {
+      return null;
+    }
+  }
+
+  if (
+    response.statusCode === 429 ||
+    response.statusCode === 418 ||
+    response.statusCode === 413 ||
+    (response.statusCode >= 500 && response.statusCode < 600)
+  ) {
```

Comment:
```suggestion
  if (response.statusCode < 200 || response.statusCode > 300) {
```

Reply:
I don't think we want to retry 404 errors or 401 errors which is why Got is selective. This matches their defaults

Reply:
Ok, I see that problem here is that `maybeRetryMS()` can actually be called for status codes that got does not retries due its usage in GraphQL requests, but in this case we should be more stringent also on 5xx. See https://github.com/sindresorhus/got/tree/v11.8.5#retry.